src/FirstSteps/RPG/Heroes/Elf.cs

```diff
@@ -52,14 +53,29 @@ public void Heal()
 
         public void CollectArrows(List<Arrow> arrows)
         {
+
+            _equipment.Backpack.AddRange(arrows.ToList());
             _arrowsBag.AddRange(arrows);
         }
 
         public override int DealDamage()
         {
+
             bool magicBowIsInTheBackpack = _equipment.Backpack.Any(x => x is MagicBow);
+            bool arrowsAreInTheBackpack = _equipment.Backpack.Any(y => y is Arrow);
+
+            if (magicBowIsInTheBackpack && arrowsAreInTheBackpack)
+            {
+                MagicBow magicBow = _equipment.Backpack.OfType<MagicBow>().FirstOrDefault();
```

Comment:
You don't need the `magicBow` object, you already check if it is available here: `magicBowIsInTheBackpack`.